Remove Quirk shouldAvoidResizingWhenInputViewBoundsChange for twitter.com

aikocny opened the pull request:
#### c82252d434c9e75795fb56cf2e6e81992a4ec524
<pre>
Remove Quirk shouldAvoidResizingWhenInputViewBoundsChange for twitter.com

<a href="https://bugs.webkit.org/show_bug.cgi?id=255268">https://bugs.webkit.org/show_bug.cgi?id=255268</a>
<a href="https://rdar.apple.com/107851720">rdar://107851720</a>

Reviewed by Brent Fulgham.

Since twitter.com has moved to x.com,we performed
additional validation to ensure whether current quirks
are valid or not. The landing page has also changed
since then.

This patch removed one of the quirks that is not needed,
we tested on iPad 11-inch with external keyboard, landscape orientation.
We went to x.com, remote inspect from MacBook,
disble Site Specific Hacks, and tap in the fields.

We no longer see any UI issues (flashing or twitching)
with and without the quirk.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAvoidResizingWhenInputViewBoundsChange const):

Canonical link: <a href="https://commits.webkit.org/281556@main">https://commits.webkit.org/281556@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/0edaf9cd528b9ea6bc149ca5e0e1a84273f7a19e

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/59840 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/39187 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/12370 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/63757 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/10364 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/61969 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/46840 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/10569 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/48525 "Passed tests") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/7252 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/61870 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/36547 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/51818 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/29369 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/33255 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/9056 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/9287 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/55185 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/9334 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/65487 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/87/builds/3768 "Built successfully") | [❌ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/9320 "Found 1 new test failure: fast/scrolling/mac/scrollbars/select-overlay-scrollbar-hovered.html (failure)") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/55866 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/86/builds/3779 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/51796 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/56006 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/13386 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/88/builds/3131 "Passed tests") | | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/34999 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/36082 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/37168 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/35827 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

brentfulgham reviewed (approved):
Looks great! Thank you for looking into this.

aikocny committed:
Remove Quirk shouldAvoidResizingWhenInputViewBoundsChange for twitter.com

https://bugs.webkit.org/show_bug.cgi?id=255268
rdar://107851720

Reviewed by Brent Fulgham.

Since twitter.com has moved to x.com,we performed
additional validation to ensure whether current quirks
are valid or not. The landing page has also changed
since then.

This patch removed one of the quirks that is not needed,
we tested on iPad 11-inch with external keyboard, landscape orientation.
We went to x.com, remote inspect from MacBook,
disble Site Specific Hacks, and tap in the fields.

We no longer see any UI issues (flashing or twitching)
with and without the quirk.

* Source/WebCore/page/Quirks.cpp:
(WebCore::Quirks::shouldAvoidResizingWhenInputViewBoundsChange const):

Canonical link: https://commits.webkit.org/281556@main
